folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+    return getTranslation(locale, getDefaultTranslation().withLocale(locale));
+  }
+
+  /**
+   * Find a TranslationMap for the default locale -- used to initialize for {@code getDefaultTranslations}.
+   *
+   * @return the best applicable translation
+   * @throws IllegalStateException if no translation can be found
+   */
+  protected TranslationMap resolveDefaultLocale() {
+    TranslationMap foundDefault = getTranslation(
+      configuration.getDefaultLocale(),
+      null
+    );
+    if (foundDefault == null) {
+      foundDefault = getTranslation(Locale.ENGLISH, null);
+    }
+    if (foundDefault == null) {
+      throw new IllegalStateException(
+        String.format(
+          "No translations are sufficient for the server's default locale (%s) nor %s",
+          configuration.getDefaultLocale(),
+          Locale.ENGLISH
+        )
+      );
+    }
+    return foundDefault;
+  }
+
+  /**
+   * Get the translation map for the default locale (or en-us, if that is not possible).
+   *
+   * @return the default locale's translation map
+   */
+  public TranslationMap getDefaultTranslation() {
+    // computeIfAbsent does not work due to the resolver potentially filling multiple keys
+    if (
+      !this.localeTranslations.containsKey(configuration.getDefaultLocale())
+    ) {
+      this.localeTranslations.put(
+        configuration.getDefaultLocale(),
+        this.resolveDefaultLocale()
+      );
+    }
+    return this.localeTranslations.get(configuration.getDefaultLocale());
+  }
+
+  /**
+   * Get all of the locales for the current request.  This will respect the Accept-Language header
+   * and quality values, as applicable, with highest quality first.
+   *
+   * @return a List of all Locale objects.  Invalid and wildcard Locales will likely return a Locale
+   *   with empty string/fields or, in some cases, the server's current Locale.
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language">MDN docs</a>
+   */
+  @SuppressWarnings("java:S1166")
+  public List<Locale> getCurrentLocales() {
+    try {
+      return Collections.list(
+        (
+          (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()
+        ).getRequest()
+          .getLocales()
+      );
+    } catch (IllegalStateException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Get the best single locale.  Either the first from the request or the server's default.
+   *
+   * @return a single Locale object.  Invalid and wildcard Locales sent to the server will likely
+   *   return a Locale with empty string/fields or, in some cases, the server's current Locale.
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language">MDN docs</a>
+   */
+  @SuppressWarnings("java:S1166")
+  public Locale getCurrentLocale() {
+    try {
+      return getCurrentLocales().get(0);
+    } catch (RuntimeException e) {
```

Comment:
This `catch` seems overly broad, considering the method just swallows it. Is there a more specific exception or set of exceptions to catch instead? At the very list, it should probably logged or something